pkgs/development/compilers/gcc/common/libgcc.nix

```diff
@@ -12,7 +12,7 @@ in
   outputs = previousAttrs.outputs ++ lib.optionals enableLibGccOutput [ "libgcc" ];
   # This is a separate phase because gcc assembles its phase scripts
   # in bash instead of nix (we should fix that).
-  preFixupPhases = (previousAttrs.preFixupPhases or []) ++ [ "preFixupLibGccPhase" ];
+  preFixupPhases = (previousAttrs.preFixupPhases or []) ++ lib.optionals ((!langC) || enableLibGccOutput) [ "preFixupLibGccPhase" ];
```

Comment:
Why was this particular line added?

The other line is definitely needed; thank you for adding that one.

But this line is already wrapped in a `lib.optionalAttrs`, and (unless I made a mistake) if you expand out the definition of `enableLibGccOutput` this conditional is never false when the enclosing conditional is true.

Running a build now to see if this conditional can be dropped.

Reply:
This is added to avoid `preFixupLibGccPhase` being empty. `(!langC) || enableLibGccOutput` should be false here, since `langJit` is true for `gccgo`.

Reply:
Ah thanks, I get it now.  I need to refactor these conditionals, it's gotten to the point where even I don't remember why some of them are there, which is really bad.